allauth/socialaccount/providers/base.py

```diff
@@ -24,6 +24,17 @@ class AuthError(object):
 
 
 class Provider(object):
+    alt_text = None
+    display_name = None
+    name = None
```

Comment:
I don't really think the above 3 attributes should be defined here.

The first two will always be set with `__init__`.

Not having `name` will give a helpful exception if someone adds a new provider without the `name` attribute specified. An improvement on both might be leaving `name = None` but then raising an exception if it isn't specified by the new Provider.